### PR TITLE
Vegetable Image Classification

### PR DESCRIPTION
**Pull request by aditya0929**
# Pull Request for DL-Simplified 💡

## Issue Title : Vegetable Image Classification #282 

- **Info about the related issue (Aim of the project)** :   predict vegetable image labels to correct classes as shown in the dataset
- **Name:** Aditya Narayan Jha
- **GitHub ID:** https://github.com/aditya0929
- **Email ID:** [adityajha8906@gmail.com](mailto:adityajha8906@gmail.com)
- **Identify yourself:** `SSOC-2023 Participant`

Closes: #282 

### Describe the add-ons or changes you've made 📃

I chose convolutional neural networks for the multi instance classification situation since the dataset included contained 15 different classes of vegetables and every class had to be taken care off so that the overall accuracy increases of the model .

First, I tried training the data across a variety of numbers of epochs until a satisfactory level of accuracy was achieved using a basic CNN with maxpooling and the addition of additional layers as needed.

I used the same strategy to train MobileNet and ResNet50V2, two additional models 

ResNet-50v2: It's an improved version of ResNet-50, using residual connections to address the vanishing gradient problem. It has 50 layers and performs well in image classification tasks.

MobileNet: It's designed for mobile and embedded devices with limited resources. MobileNet reduces model size and complexity using depth-wise separable convolutions. It's efficient and suitable for real-time applications.

Both ResNet-50v2 and MobileNet have their strengths and are suitable for different scenarios depending on the available resources and the desired trade-offs between accuracy and computational efficiency.

I have also included the learning tools that were very helpful to me in developing the model and that include a lot more in-depth information.

In the models folder , i have added comparioson.ipynb file which contains different graphical representation to compare between accuracy percentages of the models 

## Type of change ☑️

What sort of change have you made:

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ⚙️
To keep a clean code and concentrate on one model at a time, I decided to take the approach of creating separate ipynb files for each models.  Aside from the model architecture, all models' code adheres to the same process for all other aspects. 

I tested it by displaying the labels of the original images alongside the predicted labels by the model to check whether both of them are same or not and majority of the time due to the high level of accuracy the test labels and predicted labels came to be the same .

## Checklist: ☑️

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

p.s. I am attaching images for **evaluation** . 

image

image
image

### CODE REVIEW

**Review by abhisheks008 — APPROVED** [review]
Your PR is approved and ready to be merged.
@aditya0929
